BLSQ/iaso
# POLIO-1770, POLIO-1768:  delete unused scopes + migration

What problem is this PR solving? Explain here in one sentence.

Related JIRA tickets : POLIO-1770, POLIO-1768

## Self proofreading checklist


## Doc
N/A

## Changes

- POLIO-1768: migration
    - Add migration that deletes unused scopes
    - Note: rounds w/o campaign were not deleted
    
- POLIO-1770: update API
    - Check existing scopes:
        - if campaign switches to round scope, delete campaign scope and vice-versa
        - if scope per campaign, ignore scopes per round received from front end and vice versa
        - Add tests    

## How to test

On polio DB:

- Go to campaigns
- Open a campaign
- Go to Scopes tab
- Tick "separate scopes per round" box
- Add new scope
- Save

--> The old scopes should have been removed from GET response, but will still be visible in front-end because of formik state. Reloading the page will make it diappear

- Repeat the operation the other way around

## Print screen / video


Uploading Screen Recording 2024-11-29 at 15.13.32.mov…



## Notes

MERGES INTO https://github.com/BLSQ/iaso/pull/1830

## Follow the Conventional Commits specification

The **merge message** of a pull request must follow the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/) specification.

This convention helps to automatically generate release notes.

Use lowercase for consistency.

[Example](https://github.com/BLSQ/iaso/commit/8b8d7d3064138c1e57878f17b4eb922516ab0112):

```
fix: empty instance pop up

Refs: IA-3665
```

Note that the Jira reference is preceded by a _line break_.

Both the line break and the Jira reference are entered in the _Add an optional extended description…_ field.
